tests/tracking/test_model_registry.py

```diff
@@ -252,6 +252,138 @@ def test_set_registered_model_tag_with_empty_string_as_value(client):
     assert {"tag_key": ""}.items() <= client.get_registered_model(name).tags.items()
 
 
+@pytest.mark.parametrize(
+    ("filter_string", "filter_func"),
+    [
+        (None, lambda mv: True),
+        ("", lambda mv: True),
+        ("name LIKE '%2'", lambda mv: mv.name.endswith("2")),
+        ("name ILIKE '%rm%00%'", lambda mv: "00" in mv.name),
+        ("name LIKE '%rm%00%'", lambda mv: False),
+        ("name = 'badname'", lambda mv: False),
+        ("name = 'CreateRMsearchForMV03'", lambda mv: mv.name == "CreateRMsearchForMV03"),
+    ],
+)
+def test_search_model_versions_filter_string(
```

Comment:
It's a bit challenging for me to understand what's going on in this test. Can we remove parametrization? Do we need to check pagination?

Reply:
I'll update this test in a follow-up.